### PR TITLE
[Silabs] BLE bug fixes for Wi-Fi devices 

### PR DESCRIPTION
**Pull request by shgutte**
Problem
1. 917 BLE does not check if we are already subscribed In the HandleTXCharCCCDWrite function
2. 917 Ble does not support Scan Response Data
3. Support Fast and Slow advertising for 917 BleManagerImpl 

Fixes added
1. Added support for checking if we are already subscribed In the HandleTXCharCCCDWrite function
2. Added support for Scan Response Data
3. Added support for fast and slow advertising for 917 BleManagerImpl 

Testing
Tested with all combinations of RS9116 and Siw917

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -96,6 +96,9 @@ void sl_ble_init()
     //  initializing the application events map
     rsi_ble_app_init_events();
     memcpy(randomAddrBLE, &randomAddr, 6);
+    ChipLogError(DeviceLayer, "%s: BLE : randomAddr %" PRIu64 "\n", __func__, randomAddr);
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:99]: Those aren't errors. Used `ChipLogDetails` or `ChipLogProgress`

You can drop the` __func__ ` it isn't too useful here.

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -589,6 +590,21 @@ CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
...
+    responseData[index++] = ShortUUID_CHIPoBLEService[1];
+
+    responseData[index++] = static_cast<uint8_t>(mDeviceNameLength + 1); // length
+    responseData[index++] = CHIP_ADV_DATA_TYPE_NAME;                     // AD type : name
+    memcpy(&responseData[index], mDeviceName, mDeviceNameLength);        // AD value
+    index += mDeviceNameLength;
+
+    rsi_ble_set_scan_response_data(responseData, index);
 
     // err = MapBLEError(result);
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:609]: remove commented code if it is not used

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 3]
```diff
@@ -792,20 +820,31 @@ void BLEManagerImpl::HandleTXCharCCCDWrite(rsi_ble_event_write_t * evt)
...
+        // If indications are not already enabled for the connection...
+        if (!bleConnState->subscribed)
         {
-            event.Type                    = DeviceEventType::kCHIPoBLESubscribe;
-            event.CHIPoBLESubscribe.ConId = 1; // TODO:: To be replaced by device mac address
-            err                           = PlatformMgr().PostEvent(&event);
+            bleConnState->subscribed = 1;
+            // Post an event to the CHIP queue to process either a CHIPoBLE Subscribe or Unsubscribe based on
+            // whether the client is enabling or disabling indications.
+            {
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:829]: bracket pair here and the indentation are not needed.

**Review by jmartinez-silabs — COMMENTED** [review]
just some nits to address

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 4]
```diff
@@ -114,9 +117,6 @@ void sl_ble_event_handling_task(void)
     while (1)
     {
         //! This semaphore is waiting for next ble event task
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:119]: move the comment where. you moved the `rsi_semaphore_wait` call.

Also why is it moved only to the default? Can there be a ble event without the semaphore being posted first?

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 1 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: Removed the logs

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 2 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: removed

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 3 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: removed bracket

File: src/platform/silabs/rs911x/rsi_ble_config.h [hunk 5]
```diff
@@ -24,6 +24,11 @@
  * *                      Macros
  * ******************************************************/
 //! application event list
+#ifndef RSI_FAILURE
+// failure return value
+#define RSI_FAILURE -1
+#endif
```
Comment (mkardous-silabs) [inline src/platform/silabs/rs911x/rsi_ble_config.h:30]: why is the `ifndef` necessary? seems odd that have this line?

Comment (mkardous-silabs) [reply src/platform/silabs/rs911x/rsi_ble_config.h:30]: Can you add a comment with a TODO to remove these lines when we transition to the new SDK?

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 4 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: @jmartinez-silabs added this in JIRA i'll check and update this in upcoming PR
https://jira.silabs.com/browse/MATTER-2184

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**